### DISCUSSION TITLE
How come posts from my bot seem unsearchable?

### DISCUSSION THREAD

**Discussion by bgawalt** [bluesky-social/atproto#1950] [Clarifications and "Why?"]
Hi everyone! I have a little Python routine that autoposts "What Brian's Reading These Days" updates at pseudorandom intervals, onto my personal account:

* Example post: [here](https://bsky.app/profile/brian.gawalt.com/post/3kg5szhk3rw2k)
* The script that's a-postin': [readerbot_atp.py](https://github.com/bgawalt/reader_bot/blob/master/readerbot_atp.py) 

It always includes a hashtag, "[hash]ReaderBot" (which for fun, I rich-text-facet'ify as a hyperlink to https://bsky.app/search?q=%23ReaderBot ).

Here's my question: I can't actually get BSky.App's search to pull up any of those bot-authored posts, whether I search for the hashtag, or any other words that show up in the autoposted messages. Does my bot omit some field that says, "yes, please make this post searchable"?

Thanks much, (and apologies if this is too bsky.app-centric for a venue that's devoted to the atproto overall,)
--Brian

**Comment by DavidBuchanan314** [comment]
It looks like you're not setting the (optional) `lang` field in the post record. Ideally that shouldn't matter, but I think that might be what's breaking searchability.

**Reply by bgawalt** [reply]
This was a good idea! I tried it out:

*  Latest post, presumably with `langs` attached: [here](https://bsky.app/profile/brian.gawalt.com/post/3kgaaw4bsjs26)
* Code update that's now attaching `langs = [en-US]`: [here](https://github.com/bgawalt/reader_bot/commit/9231040d9e6d39238302cd27d8768dbcb73550ba)

but, sadly, still seems like they're not getting indexed for search: https://bsky.app/search?q=Brian

Zooming out -- are everyone/anyone else's bots getting indexed like normal posts typed by hand into a normal client?

**Comment by bgawalt** [comment]
Hey, I went quiet on this, but: it fixed itself! https://bsky.app/search?q=ReaderBot now surfaces all the posts my bot's ever made to my account.

**Reply by bgawalt** [reply]
(It fixed itself a while ago, maybe around app v 1.58? 1v1.57?)
